src/transformState.js

```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
-}
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+
+        break;
+
+      case 'removeProperties':
+        for (const keys of action.keysToRemove) {
+          delete state[keys];
+        }
+
+        break;
 
+      case 'clear':
+        for (const keys in state) {
+          delete state[keys];
+        }
+
+        break;
+
+      default:
+        return 'Unknown command';
```

Comment:
`switch/case should always have default case for error handling.` (Checklist p4)

Let's try `throw new Error()`. In which we describe the error and the action.type that fell into default